iosApp/Survey/Sources/Presentation/Modules/SurveyDetail/SurveyQuestion/QuestionEmojiView.swift

```diff
@@ -56,7 +56,7 @@ struct QuestionEmojiView: View {
             Spacer()
         }
         .onChange(of: rating) {
-            answers = [options[$0].id]
+            answers = [.init(id: options[$0].id, answer: nil)]
```

Comment:
It could be safer to check for `rating` validity before accessing the `option` array 💭
```suggestion
            guard $0 < options.count else { return }
            answers = [.init(id: options[$0].id, answer: nil)]
```